pandas/tests/extension/test_categorical.py

```diff
@@ -223,6 +223,28 @@ def test_cast_category_to_extension_dtype(self, expected):
 
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize(
+        "data, dtype, expected",
+        [
+            (
+                "2015-01-01",
+                "datetime64[ns]",
+                np.array(["2015-01-01T00:00:00.000000000"], dtype="datetime64[ns]"),
+            ),
+            (
+                "2015-01-01",
+                "datetime64[ns, MET]",
+                np.array(
+                    [Timestamp("2015-01-01 00:00:00+0100", tz="MET")], dtype=object
+                ),
+            ),
+        ],
+    )
+    def test_consistent_casting(self, data, dtype, expected):
+        # GH 28448
```

Comment:
The OP uses `asarray` which is different than what is being tested here. Can you not do `asarray` in the test? The return type should be a date time array

Reply:
As in, `pandas.arrays.DatetimeArray`? `pd.Categorical(data).astype(dtype)` is only of that type in the second case (dtype datetime64[ns, MET]).

In the first case (dtype datetime64[ns]), it is of type `np.array`.

Reply:
Hmm well I think that is a problem that as type would return a numpy array for "datetime64[ns]" and a DatetimeArray for "datetime64[ns, <TZ>]" but @jbrockmendel or @TomAugspurger might know more

Reply:
Right I think so too. But the problem is `Categorical.astype("datetime64[ns]")` returns a NumPy array, so tz-naive date times would be a numpy array where tz-aware would be a DTA.

That seems odd to me but I don't know the entire history of these

Reply:
In theory `Categorical.astype('datetime64[ns]')` could return a tz-naive DatetimeArray. I'm not sure if it was discussed explicitly when DatetimeArray was added, but changing that would be API-breaking and IMO not worth it.

Reply:
@WillAyd I tend to agree with everyone above in that this is sub-optimal behavior but not worth changing on its own\*.  Is this a problem or A Problem?

\* There's been some discussion of making EA.astype _always_ return EA at some point, which I think would include this.

Reply:
If you guys don't think so then no. I just wasn't aware of the historical context hence the ping - thanks for the insights.

So @MarcoGorelli still should update the tests to use `asarray` but vary expectation accordingly

Reply:
Ok, thanks all for your feedback - I've updated the tests